### PR TITLE
Day night cycle

### PR DESCRIPTION
**Pull request by tobihoffmann**
changes in GameScene:
- removed diretional light
- added prefab Gametime (contains directional lights for sun/moon

test this by changing Day Length in Clock.cs on GameTime.prefab or dragging its Time Of Day Slider

### CODE REVIEW

File: Assets/Scripts/GameTime/Clock.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            UpdateTimeScale();
+            UpdateTime();
+        }
+
+        /// <summary>
+        /// Updates the current timeScale; Varies depending on the current time of day (=> timeCurve)
+        /// </summary>
+        private void UpdateTimeScale()
+        {
+            _timeScale = 24 / (dayLength / 60);
```
Comment (thuemmels) [inline Assets/Scripts/GameTime/Clock.cs]: seems like this calculation is constant, so maybe move that out of the update method to save resources

**Review by thuemmels — CHANGES_REQUESTED** [review]

File: Assets/Scripts/GameTime/Clock.cs (hunk 1 above)
Comment (tobihoffmann) [reply Assets/Scripts/GameTime/Clock.cs]: good idea

File: Assets/Scripts/GameTime/Clock.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            _timeScale *= timeCurve.Evaluate(timeOfDay);
+            _timeScale /= _timeCurveNormalization;
+        }
+
+        /// <summary>
+        /// Updates the current time based on FPS and timeScale
+        /// </summary>
+        private void UpdateTime()
+        {
+            timeOfDay += Time.deltaTime * _timeScale / 86400;
```
Comment (DavidPgl) [inline Assets/Scripts/GameTime/Clock.cs:58]: What's 8640?

File: Assets/Scripts/GameTime/DayNightCycle.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        [SerializeField][Tooltip("A gradient that changes the sun color depending of day time")]
+        private Gradient sunColor;
+
+        [SerializeField][Tooltip("tilt of the sun (and moon)")] [Range(-45f, 45f)]
+        private float tilt; // could vary depending seasons in the future
+        
+        private float _intensity;
+
+        void Update()
```
Comment (DavidPgl) [inline Assets/Scripts/GameTime/DayNightCycle.cs]: Access modifier

**Review by DavidPgl — DISMISSED** [review]

File: Assets/Scripts/GameTime/Clock.cs (hunk 2 above)
Comment (tobihoffmann) [reply Assets/Scripts/GameTime/Clock.cs:58]: the amount seconds in 24 hours

File: Assets/Scripts/GameTime/DayNightCycle.cs (hunk 3 above)
Comment (tobihoffmann) [reply Assets/Scripts/GameTime/DayNightCycle.cs]: shame over my main

**Review by Kaes3kuch3n — DISMISSED** [review]

**Comment by Kaes3kuch3n** [comment]
Looks good after fixing the stuff from the comments, I'd say :)

File: Assets/Scripts/GameTime/Clock.cs (hunk 1 above)
Comment (SamuelllG) [reply Assets/Scripts/GameTime/Clock.cs]: you moved it from UpdateTimeScale to Update? 😀

File: Assets/Scripts/GameTime/Clock.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        private void Awake()
+        {
+            NormalizeTimeCurve();
+        }
+
+        private void Update()
+        {
+            UpdateTimeScale();
+            UpdateTime();
+            _timeScale = 24 / (dayLength / 60); // calculating basic tick rate
```
Comment (SamuelllG) [inline Assets/Scripts/GameTime/Clock.cs]: you moved it from UpdateTimeScale to Update? 😀

**Review by SamuelllG — DISMISSED** [review]
Sun move, I approve
Sieht nice aus 👌

File: Assets/Scripts/GameTime/Clock.cs (hunk 4 above)
Comment (Kaes3kuch3n) [reply Assets/Scripts/GameTime/Clock.cs]: Could you please move this to Awake or to the field declaration @tobihoffmann? :)

**Review by Kaes3kuch3n — CHANGES_REQUESTED** [review]

File: Assets/Scripts/GameTime/Clock.cs (hunk 4 above)
Comment (tobihoffmann) [reply Assets/Scripts/GameTime/Clock.cs]: oh wow. :D

**Review by SamuelllG — APPROVED** [review]

**Review by inkompotato — APPROVED** [review]
beautiful, once merged, time of day will be added to savegames

**Pull request merged**